lib/stamp/emitters/composite.rb

```diff
@@ -7,12 +7,9 @@ def initialize
         @emitters = []
       end
 
-      def format(target)
-        # NOTE using #each to build string because benchmarking shows
-        # that it's ~20% faster than .map.join('')
-        result = ''
-        @emitters.each { |e| result << e.format(target).to_s }
-        result
+      def format(out, target)
```

Comment:
I'm not sure why this is better. It seems like it just distributes string-building repsonsibility to the emitters.